org.OpenT2T.Sample.SuperPopular.Thermostat/Wink Thermostat/v2/thingTranslator.js

```diff
@@ -0,0 +1,88 @@
+/*jshint esversion: 6 */
+/* jshint node: true */
+"use strict";
+var wh = require("opent2t-translator-helper-wink");
+
+// This code uses ES2015 syntax that requires at least Node.js v4.
+// For Node.js ES2015 support details, reference http://node.green/
+
+function validateArgumentType(arg, argName, expectedType) {
+    if (typeof arg === 'undefined') {
+        throw new Error('Missing argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + '.');
+    } else if (typeof arg !== expectedType) {
+        throw new Error('Invalid argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + ', got: ' + (typeof arg));
+    }
+}
+
+// This translator class implements the "org.OpenT2T.Sample.SuperPopular.Thermostat" interface.
+class WinkThermostat {
+
+    constructor(deviceInfo) {
+        this.initDevice(deviceInfo);
+    }
+
+    initDevice(deviceProps) {
+        console.log('Initializing device.');
+        var device = deviceProps;
+
+        validateArgumentType(device, 'device', 'object');
+        validateArgumentType(device.props, 'device.props', 'string');
+
+        var props = JSON.parse(device.props);
+        validateArgumentType(props.access_token, 'device.props.access_token', 'string');
+        validateArgumentType(props.id, 'device.props.id', 'string');
+       
+        var deviceId = props.id; 
+        var accessToken = props.access_token;
+
+        //use the winkHub device and initialize
+        wh.initWinkApi("thermostats",deviceId,accessToken);
+        console.log('Javascript and Wink Helper initialized : ' + deviceId + accessToken);
+    }
+
+    get CurrentTemperature() {
+        console.log('Get current temperature.');
+        return wh.getLastReading('temperature'); 
+    }
+
+    get HeatingSetpoint() {
+        console.log('Get heating point.');
+        return wh.getValueOfDesiredState('min_set_point');
+    }
+
+    set HeatingSetpoint(targetValue) {
+         console.log("Set Heating Setpoint");
+        return wh.sendDesiredStateCommand('min_set_point',targetValue);
+    }
+
```

Comment:
Sorry if I wasn't clear: These need to be changed back to normal getter/setter methods.

There is an important semantic difference between these two declarations:
```
setHeatingSetpoint(targetValue) { ... }
set HeaintSetpoint(targetValue) { ... }
```
The first can return a value (a Promise in this case). For the second, the caller has no way to access the return value, because they invoke it like this:
```
translator.HeatingSetPoint = value;
```
Because of that, callers wouldn't be able to properly wait for async set operations.